projects/schematics/src/migrations/2_0/ssr/ssr.ts

```diff
@@ -0,0 +1,298 @@
+      'compile:server',
+      'build:ssr',
+      'serve:ssr',
+      'build:client-and-server-bundles',
+    ];
+    const packagesToAddOrUpdate: NodeDependency[] = [
+      {
+        type: NodeDependencyType.Default,
+        version: '^9.0.0',
+        name: '@nguniversal/express-engine',
+        overwrite: true
+      },
+      {
+        type: NodeDependencyType.Dev,
+        version: '^9.0.0',
+        name: '@nguniversal/builders',
+        overwrite: true
+      },
+    ];
+    const projectName = await getProjectNameFromWorkspace(tree);
+
+    scripts.forEach(key => {
+      const keyBackup = `${key}_bak`;
+      const scriptValue = packageJson.scripts[key];
+      if (scriptValue && !packageJson.scripts[keyBackup]) {
+        packageJson.scripts[keyBackup] = scriptValue;
+        packageJson.scripts[key] = undefined;
+      }
+    });
+
+    packageJson.scripts['dev:ssr'] = `ng run ${projectName}:serve-ssr`;
+    packageJson.scripts['serve:ssr'] = `node dist/${projectName}/server/main.js`;
+    packageJson.scripts['build:ssr'] = `ng build --prod && ng run ${projectName}:server:production`;
+    packageJson.scripts['prerender'] = `ng run ${projectName}:prerender`;
+
+    tree.overwrite(pkgPath, JSON.stringify(packageJson, null, 2));
+
+    removePackageJsonDependency(tree, '@nguniversal/module-map-ngfactory-loader');
+    packagesToAddOrUpdate.forEach((dep) => {
+      addPackageJsonDependency(tree, dep);
+    });
+
+  };
+}
+
+export function removeImportsInMainServerFile() {
+  return (tree: Tree, context: SchematicContext) => {
+    context.logger.info(
+      'Removing unnecessary ngExpressEngine import after migration'
+    );
+    const mainServerTsPath = 'src/main.server.ts';
+    const buffer = tree.read(mainServerTsPath);
+
+    if (!buffer) {
+      throw new SchematicsException('Could not find main.server.ts');
+    }
+
+    const exportLineToRemove = "export const ngExpressEngine = NgExpressEngineDecorator.get(engine);";
+    const fileContent = buffer.toString();
+    const recorder = tree.beginUpdate(mainServerTsPath);
+    const mainServerTsSourceFile = createSourceFileWithStrippedBOM(tree, mainServerTsPath);
+
+    removeFullLineFromExportDeclarationsByModuleName(mainServerTsSourceFile, recorder, '@nguniversal/module-map-ngfactory-loader');
+    removeFullLineFromImportDeclarationsByModuleName(mainServerTsSourceFile, recorder, '@nguniversal/express-engine');
+    removeFullLineFromImportDeclarationsByModuleName(mainServerTsSourceFile, recorder, '@spartacus/core');
+    recorder.remove(fileContent.indexOf(exportLineToRemove), exportLineToRemove.length);
+
+    tree.commitUpdate(recorder);
+  };
+}
+
+export function removeMapLoaderModule() {
+  return (tree: Tree, context: SchematicContext) => {
+    context.logger.info(
+      'Removing MapLoaderModule import'
+    );
+
+    const mapLoaderModuleTsPath = 'src/app/app.server.module.ts';
+    const buffer = tree.read(mapLoaderModuleTsPath);
+
+    if (!buffer) {
```

Comment:
As `buffer` is not used after this line, maybe we check directly for `if(tree.read(mapLoaderModuleTsPath))`?

Reply:
Yea, good idea